config/file_config_test.go

```diff
@@ -95,12 +96,35 @@ func TestSetFederationNodes(t *testing.T) {
 	keyPair := keys.Ed25519KeyPairForTests(0)
 
 	node1 := &hardCodedFederationNode{
-		nodePublicKey:  keyPair.PublicKey(),
+		nodePublicKey: keyPair.PublicKey(),
+	}
+
+	require.EqualValues(t, node1, cfg.FederationNodes(0)[keyPair.PublicKey().KeyForMap()])
+}
+
+func TestSetGossipPeers(t *testing.T) {
+	cfg, err := NewEmptyFileConfig(FILE_CONFIG_CONTENTS)
```

Comment:
You have several tests all sharing the same config stub. What might happen is that all tests fail on parse error if any of the parse method is broken. I'd prefer creating an inline stub for the minimal config required by each individual test